subsys/net/ip/tcp2.c

```diff
@@ -468,16 +474,18 @@ static uint8_t *tcp_options_get(struct net_pkt *pkt, int tcp_options_len)
 static bool tcp_options_check(struct tcp_options *recv_options,
 			      struct net_pkt *pkt, ssize_t len)
 {
+	uint8_t options_buf[40]; /* TCP header max options size is 40 */
 	bool result = len > 0 && ((len % 4) == 0) ? true : false;
-	uint8_t *options = tcp_options_get(pkt, len);
+	uint8_t *options = tcp_options_get(pkt, len, options_buf,
+					   sizeof(options_buf));
 	uint8_t opt, opt_len;
 
 	NET_DBG("len=%zd", len);
 
 	recv_options->mss_found = false;
 	recv_options->wnd_found = false;
 
-	for ( ; len >= 1; options += opt_len, len -= opt_len) {
+	for ( ; options && len >= 1; options += opt_len, len -= opt_len) {
```

Comment:
Is options == NULL an error condition?  If so, log it and return an error.

Reply:
We are already printing warning at the end of the function if the options are not set properly, no need to print any extra info here.